ldap/servers/plugins/memberof/memberof.c

```diff
@@ -813,16 +811,25 @@ memberof_call_foreach_dn(Slapi_PBlock *pb __attribute__((unused)), Slapi_DN *sdn
             if (memberof_entry_in_scope(config, base_sdn)) {
                 /* do nothing, entry scope is spanning
                  * multiple suffixes, start at suffix */
-            } else if ((scope_sdn = memberof_scope_is_child_of_dn(config, base_sdn))) {
-                /* scope is below suffix, set search base */
-                base_sdn = scope_sdn;
+            } else if (config->entryScopes) {
+                for (size_t i = 0; config->entryScopes[i]; i++) {
+                    if (slapi_sdn_issuffix(config->entryScopes[i], base_sdn)) {
+                        /* Search each include scope */
+                        slapi_search_internal_set_pb(search_pb, slapi_sdn_get_dn(config->entryScopes[i]),
+                                                     LDAP_SCOPE_SUBTREE, filter_str, 0, 0, 0, 0,
+                                                     memberof_get_plugin_id(), 0);
+                        slapi_search_internal_callback_pb(search_pb, callback_data, 0, callback, 0);
+                    }
+                }
+                if (!all_backends) {
+                    goto done;
+                }
```

Comment:
in case 'all_backend'==True, is there a risk that it goes to the internal search below (line 836). In such case is base_sdn valid and is it processing the base_sdn twice ?

Reply:
This function is actually rather confusing.  I've revised it again, and it passes CI tests, but please review...

Reply:
@tbordaz could you review this again?  Remember there is second part to this that is waiting for this to be merged first.